gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py

```diff
@@ -4,537 +4,197 @@
-        hm = {
-            "admin_state_up": True,
-            "tenant_id": "f6b09b7a590642d8ac6de73df0ab0686",
-            "delay": 10,
-            "max_retries": 3,
-            "timeout": 10,
-            "pools": [],
-            "type": "PING",
-                    "id": "c30d8a88-c719-4b93-aa64-c58efb397d86"
-        }
-        return hm
-
-    def _fake_kwargs(self):
-        kwargs = {'service_type': 'loadbalancer',
-                  'vm_mgmt_ip': '172.24.4.5',
-                  'mgmt_ip': '172.24.4.5',
-                  'source_cidrs': ['1.2.3.4/24'],
-                  'destination_cidr': ['1.2.3.4/24'],
-                  'gateway_ip': '1.2.3.4',
-                  'provider_interface_position': '1',
-                  'request_info': 'some_id',
-                  'rule_info': {
-                      'active_provider_mac': '00:0a:95:9d:68:16',
-                      'provider_mac': '00:0a:95:9d:68:16',
-                      'active_stitching_mac': '00:0a:95:9d:68:25',
-                      'stitching_mac': '00:0a:95:9d:68:25',
-                      'active_fip': '172.24.4.5',
-                      'fip': '172.24.4.5',
-                      'service_id': '1df1cd7a-d82e-4bbd-8b26-a1f106075a6b',
-                      'tenant_id': '6bb921bb81254b3e90e3d8c71a6d72dc'},
-                  'context': {'notification_data': 'hello'}
-                  }
-        return kwargs
-
-
-class ControllerTestCase(unittest.TestCase):
+class LBaasRpcSenderTest(unittest.TestCase):
+
+    @mock.patch(__name__ + '.FakeObjects.conf')
+    @mock.patch(__name__ + '.FakeObjects.sc')
+    def _get_configurator_rpc_manager_object(self, sc, conf):
+        cm = configurator.ConfiguratorModule(sc)
+        dmx = demuxer.ConfiguratorDemuxer()
+        rpc_mgr = configurator.ConfiguratorRpcManager(sc, cm, conf, dmx)
+        return sc, conf, rpc_mgr
+
+    def test_notifications(self):
+        sc, conf, rpc_mgr = self._get_configurator_rpc_manager_object()
+        agent = lb.LBaasRpcSender(sc)
+
+        data = "NOTIFICATION_DATA"
+        with mock.patch.object(sc, 'new_event', return_value='foo') as (
+                mock_new_event), \
+                mock.patch.object(sc, 'poll_event') as (mock_poll_event):
+
+            agent._notification(data)
+
+            mock_new_event.assert_called_with(id='NOTIFICATION_EVENT',
+                                              key='NOTIFICATION_EVENT',
+                                              data=data)
+            mock_poll_event.assert_called_with('foo')
+
+
+class LBaaSRpcManagerTest(unittest.TestCase):
 
     def __init__(self, *args, **kwargs):
-        super(ControllerTestCase, self).__init__(*args, **kwargs)
-        self.lb = FakeObjects()
+        super(LBaaSRpcManagerTest, self).__init__(*args, **kwargs)
+        self.fo = FakeObjects()
+        self.arg_dict_vip = {
+            'context': self.fo.context,
+            'vip': self.fo._fake_vip_obj(),
+            'serialize': True,
```

Comment:
Need to correct this serialize and binding_key args. This should  not be set in arg_dict but should be set in the event object. Please check the latest lb agent code